### PR TITLE
Reset colors at end of line (fixes #5324)

### LINKED ISSUE #5324: Cleanup terminal colors on quit
Not too dramatic but would be nice to restore the colors in the terminal when client or server quit.

image

### PR DESCRIPTION
**Pull request by def-**
@ChillerDragon Can you test this?

## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/base/log.cpp [hunk 1]
```diff
@@ -242,6 +242,10 @@ class CLoggerAsync : public ILogger
 			aio_write_unlocked(m_pAio, aAnsi, str_length(aAnsi));
 		}
 		aio_write_unlocked(m_pAio, pMessage->m_aLine, pMessage->m_LineLength);
+		if(m_AnsiTruecolor)
```
Comment (heinrich5991) [inline src/base/log.cpp:245]:
```suggestion
		if(m_AnsiTruecolor && pMessage->m_HaveColor)
```

**Comment by ChillerDragon** [comment]
Wowo deen being super fast again!

Yea fixed it for me. BTW I use bash and a prompt that does not set colors to test this. `PS1='$ ' ./DDNet` when quitting the client while being in game the last message is grey.

Not sure if it is needed to reset the color after every print. They way I saw it in the curses universe is usually saving all the tty info on boot and then restore on close. But yea if its not too much overhead its probably fine this way as well.

File: src/base/log.cpp (hunk 1 above)
Comment (def-) [reply src/base/log.cpp:245]: Done

**Comment by def-** [comment]
If we are killed with `kill -9`, we can't reset. If another process writes into the same terminal, for example when you run multiple processes with &, we can't reset. So I think it's better to reset after each message,

**Comment by heinrich5991** [comment]
bors r+

Thanks. :)

**Pull request merged**